src/psyclone/dynamo0p3.py

```diff
@@ -2709,15 +2710,19 @@ def gen_code(self, parent):
         self.evaluators.initialise_basis_fns(invoke_sub)
 
         if self.is_coloured():
-            # Add declarations of the colour map and array holding the
-            # no. of cells of each colour
-            invoke_sub.add(DeclGen(parent, datatype="integer",
+            # Declare the colour map
+            declns = ["cmap(:,:)"]
+            if not config.DISTRIBUTED_MEMORY:
```

Comment:
OK, done. Distributed memory calls a function for the upper bound so does not require a variable to be declared. This is the API provided by the Met Office. However, perhaps this API should be changed in the future to avoid dereferencing near parallel code (as this may cause problems with directives). I believe that this was the reason for the code being the way it is in the non-distributed memory version.